test/sslapitest.c

```diff
@@ -9831,6 +9876,255 @@ static int test_load_dhfile(void)
 #endif
 }
 
+#if !defined(OSSL_NO_USEABLE_TLS1_3) && !defined(OPENSSL_NO_COMP) && (defined(ZLIB) || !defined(OPENSSL_NO_BROTLI) || !defined(OPENSSL_NO_ZSTD))
+/*
+ * Test 0 = Skip compression
+ * Test 1 = libssl does the compression
+ * Test 2 = error out during compression
+ * Test 3 = libssl does compression, client authentication
+ * Test 4 = app does the compression
+ */
+/* This callback just says to skip compression */
+static int cert_comp_cb0(SSL *ssl, uint16_t alg, void* arg)
+{
+    *(int*)arg = alg;
+    return 0;
+}
+/* This callback just says to do the compression */
+static int cert_comp_cb1(SSL *ssl, uint16_t alg, void* arg)
+{
+    *(int*)arg = alg;
+    return alg;
+}
+/* This callback just says to error out */
+static int cert_comp_cb2(SSL *ssl, uint16_t alg, void* arg)
+{
+    return -1;
+}
+static int cert_comp_cb4(SSL *ssl, uint16_t alg, void* arg)
+{
+    unsigned char *cert_data = NULL;
+    size_t cert_length = 0;
+    unsigned char *comp_data = NULL;
+    size_t comp_length = 0;
+    int retval = 0;
```

Comment:
Should this be -1? i.e. error out on failure (not skip compression).

Reply:
Probably.